src/mappings/event/loans.ts

```diff
@@ -100,17 +102,17 @@ export async function newMarket(
         token: currency,
         height: height,
         timestamp: timestamp,
-        borrowCap: market.borrowCap,
-        supplyCap: market.supplyCap,
+        borrowCap: divideByTenToTheNth(market.borrowCap, decimals),
+        supplyCap: divideByTenToTheNth(market.supplyCap, decimals),
         rateModel: InterestRateModel,
-        closeFactor: market.closeFactor,
+        closeFactor: market.closeFactor / 10**6,
         lendTokenId: lendTokenIdNo,
         state: MarketState.Pending,
-        reserveFactor: market.reserveFactor,
-        collateralFactor: market.collateralFactor,
-        liquidateIncentive: market.liquidateIncentive,
-        liquidationThreshold: market.liquidationThreshold,
-        liquidateIncentiveReservedFactor: market.liquidateIncentiveReservedFactor,
+        reserveFactor: market.reserveFactor / 10**6,
+        collateralFactor: market.collateralFactor / 10**6,
+        liquidateIncentive: divideByTenToTheNth(market.liquidateIncentive, 18),
```

Comment:
Same as for the other cases, please use a constant instead of a magic number, and indicate in the constant name, or a comment why the number is 18.

Reply:
@daniel-savu , similar to above, do you have a suggestion for a constant name? ie. Something that indicates the origin of the number 18 as decimal factor.

Reply:
18 decimals is what's used by the parachain type (`FixedU128`). The `interbtc-api` lib does have a decoding function for this type: https://github.com/interlay/interbtc-api/blob/a629d9a846804cf207838794f368606c7ebc7a10/src/utils/encoding.ts#L122

Reply:
@mafux777 ideally, you would either use the function that Dan referenced here or define the constants similar to how the lib does it such that you have something like this at the beginning of the file and then instead of referencing a magic number, use the constant:

```ts
// https://github.com/paritytech/substrate/blob/8ae4738bd7ee57556ea42c33600dc95488b58db6/primitives/arithmetic/src/fixed_point.rs#L2200
const FIXEDI128_SCALING_FACTOR = 18;
// https://github.com/paritytech/substrate/blob/8ae4738bd7ee57556ea42c33600dc95488b58db6/primitives/arithmetic/src/per_things.rs#L1881
const PERMILL_BASE = 1000000;

reserveFactor: market.reserveFactor / PERMILL_BASE,
collateralFactor: market.collateralFactor / PERMILL_BASE,
liquidateIncentive: divideByTenToTheNth(market.liquidateIncentive, FIXEDI128_SCALING_FACTOR),
...

Reply:
Hi, I have used the constants and not the functions, because they would have required a lot of back and forth type casting that would have made the code a lot harder to read.
This particular app is now optimised to feed into an analytical dashboard, and Grafana does not like BIGINT. Therefore, I have supplied human- and grafana-friendly numbers to the squid. The dashboard reads directly from the squid and need not make any further transformations with big numbers or percentages.

Reply:
I have used the constants and not the functions, because they would have required a lot of back and forth type casting that would have made the code a lot harder to read.
This particular app is now optimised to feed into an analytical dashboard, and Grafana does not like BIGINT. Therefore, I have supplied human- and grafana-friendly numbers to the squid. The dashboard reads directly from the squid and need not make any further transformations with big numbers or percentages.